packages/@react-aria/grid/src/useGrid.ts

```diff
@@ -107,14 +108,39 @@ export function useGrid<T>(props: GridProps, state: GridState<T, GridCollection<
   });
 
   let domProps = filterDOMProps(props, {labelable: true});
+
+  let onFocus = useCallback((e) => {
+    if (manager.isFocused) {
+      // If a focus event bubbled through a portal, reset focus state.
+      if (!e.currentTarget.contains(e.target)) {
+        manager.setFocused(false);
+      }
+
+      return;
+    }
+
+    // Focus events can bubble through portals. Ignore these events.
+    if (!e.currentTarget.contains(e.target)) {
+      return;
+    }
+
+    manager.setFocused(true);
+  }, [manager]);
+
+  // Continue to track collection focused state even if keyboard navigation is disabled
+  let navDisabledHandlers = useMemo(() => ({
+    onBlur: collectionProps.onBlur,
+    onFocus
+  }), [onFocus, collectionProps.onBlur]);
+
```

Comment:
What happened here?

Reply:
Was were running into a case where calling `state.setKeyboardNavigationDisabled(true);` in TableView from the resize menu item would break grid keyboard navigation due to the timing with which the listeners from useSelectableCollection were turned off by `isKeyboardNavigationDisabled`. If we immediately turned off keyboard navigation before focus moved from clicking the column's menu item to the resizer, [this line](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/selection/src/useSelectableCollection.ts#L284) wouldn't be called and thus [useSelectableItem](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/selection/src/useSelectableItem.ts#L140) wouldn't focus the table cell as you tried to keyboard navigate

`manager.isFocused` tracking didn't feel like it should be disabled just because keyboard navigation was disabled thus the addition of the `navDisabledHandlers`